### PR TITLE
Meal foods creation

### LINKED ISSUE #13: MealFoods Joins Table Initial Schema
**REQUIRES #6 AND #12**
A MealFood resource should be stored in the database on a MealFoods table.
The structure of a MealFood resource is as follows:
```
ID: Integer
Meal_id: Integer
Food_id: Integer
```
This also sets up an `many-to-many` relationship between Foods and Meals. Their models will need to be updated accordingly.

### PR DESCRIPTION
**Pull request by tnodland**
Creates the MealFood model to make a many-to-many relationship between meals and foods.  Updates the associations on both meals and foods to have this relationship

resolves #13

### CODE REVIEW

File: models/mealfood.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+           },
+           
+    mealId: {type: DataTypes.INTEGER,
+             allowNull: false,
+             references: {
+               model: 'Meal',
+               key: 'id'
+             }
+           }
+
```
Comment (BrennanAyers) [inline models/mealfood.js]: It's strange that this works, but, Sequelize is a rabbit hole we don't need to go any deeper down for now 🐇

File: migrations/20190819222634-create-food.js [hunk 2]
```diff
@@ -9,10 +9,12 @@ module.exports = {
...
       },
       name: {
-        type: Sequelize.STRING
+        type: Sequelize.STRING,
+        allowNull: false
       },
       calories: {
-        type: Sequelize.INTEGER
+        type: Sequelize.INTEGER,
+        allowNull :false
```
Comment (BrennanAyers) [inline migrations/20190819222634-create-food.js]: Good catch!

File: models/mealfood.js (hunk 1 above)
Comment (tnodland) [reply models/mealfood.js]: The guide I followed for this was found at https://medium.com/@the_ozmic/how-to-create-many-to-many-relationship-using-sequelize-orm-postgres-on-express-677753a3edb5

File: migrations/20190824182911-create-meal.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    return queryInterface.createTable('Meals', {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER
+      },
+      name: {
+        type: Sequelize.STRING,
+        allowNull: false
```
Comment (BrennanAyers) [inline migrations/20190824182911-create-meal.js]: Thanks!

**Pull request merged**